src/core/net/srp_server.cpp

```diff
@@ -441,25 +477,96 @@ void Server::SelectPort(void)
 
 void Server::Start(void)
 {
-    Error error = kErrorNone;
-
     VerifyOrExit(mState == kStateStopped);
 
     mState = kStateRunning;
+    PrepareSocket();
+    otLogInfoSrp("[server] start listening on port %u", mPort);
```

Comment:
`Srp::Server::Stop`  needs to be updated accordingly.

Reply:
@wgtdkp I am not sure if I follow your suggestion here. Is it related to the log?

We already have this log line in the `Stop()`:
```c++
void Server::Stop(void)
{
    VerifyOrExit(mState == kStateRunning);

    [...]

    otLogInfoSrp("[server] stop listening on %u", mPort);
    IgnoreError(mSocket.Close());
    mHasRegisteredAnyService = false;

exit:
    return;
}
```

Reply:
I meant we may want to close the socket depending on whether it is sharing the DNS-SD socket. But I think it is always okay to just close `mSocket` in `::Srp::Server::Stop`.

Reply:
Yes. We can close an already closed socket again which then does nothing.